### ISSUE TITLE
SecondLevelModel is late to notice Target and Reference image mismatch

### ISSUE THREAD

**Issue by andreifoldes** [nilearn/nilearn#3298]
I think that the SecondLevelModel object when operating on a list of FirstLevelModels should first check whether all FirstLevelModels are in the same Affine and Shape before running because in my case it fails after processing multiple models.

In addition it would be nice if there was a feature to force resample everything at the 2nd level as well.

### Benefits to the change
User would know in advanced that data is not fit for 2nd level analysis prior to it starting.

### Pseudocode for the new behavior, if applicable

```python

```

**Comment by bthirion** [comment]
Thx for raising this. Indeed there are two points.
* checking a priori in indeed important
*  Regarding forced resampling, I am less enthusiastic: what referential should be taken by default ? I think that we resample safely only if something like target_affine and target_shape or target_img are provided ?

**Comment by andreifoldes** [comment]
Right - with regards to "forced resampling" - I'm only suggesting it as an additional option of course - when I faced this issue recently I chose my first image as reference, but maybe a more elegant thing would be to choose the reference of the majority of the 1st level images/FirstLevelModels?

Indeed I was hoping that by specifying a target_affine and target_shape in my SecondLevelModel I could trigger the resampling to happen however I got the following error

>   File "/home/user/anaconda3/envs/spyder-env/lib/python3.8/site-packages/nilearn/image/resampling.py", line 418, in resample_img
>     if target_shape is not None and target_affine.shape == (3, 3):
> 
> AttributeError: 'list' object has no attribute 'shape'

I haven't fully tracked it down, but my intuition is that I have multi-session/multi-run data and the resampling procedure can't deal with that.

**Comment by bthirion** [comment]
> but maybe a more elegant thing would be to choose the reference of the majority of the 1st level images/FirstLevelModels?

No, we should particularly avoid fancy behavior that are eventually hard to figure out.
To me, forcing resampling is only possible whenever target affine and shape and explicitly disclosed.

**Comment by andreifoldes** [comment]
Yes, that sounds fair. 

Should I try to write a PR for trying to fix the fact that is fails to resample multi-run data?

**Comment by bthirion** [comment]
Yes, thx for proposing !
